pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/context/AIBuilderContext.java

```diff
@@ -23,79 +24,120 @@
  */
 public class AIBuilderContext {
 
+    private AIBuildStep buildStep;
+    private AIBuildStep failedStep;
+    private String failedMessage;
+
     private final Handle handle;
-    private final String userGuid;
-    private final String operatorGuid;
+    private final AIBuilderParams params;
+
     private final long langCodeId;
-    private final String isoLangCode;
-    private final ContentStyle style;
-    private final FormActivityDef formActivityDef;
-    private final FormResponse formResponse;
+
+    private FormActivityDef formActivityDef;
+    private FormResponse formResponse;
+    private FormInstance formInstance;
+
+    private Long previousInstanceId;
 
     private final PexInterpreter interpreter = new TreeWalkInterpreter();
     private final I18nContentRenderer i18nContentRenderer = new I18nContentRenderer();
     private final Map<String, Object> rendererInitialContext = new HashMap<>();
 
-    private final Long previousInstanceId;
-
     private Map<Long, String> renderedTemplates = new HashMap<>();
 
-    private final AICreatorsFactory creatorsFactory;
+    private final AICreatorsFactory creatorsFactory = new AICreatorsFactory();
 
-    public AIBuilderContext(
-            Handle handle,
-            String userGuid,
-            String operatorGuid,
-            String isoLangCode,
-            ContentStyle style,
-            FormActivityDef formActivityDef,
-            FormResponse formResponse) {
+    public AIBuilderContext(Handle handle, AIBuilderParams params) {
         this.handle = handle;
-        this.userGuid = userGuid;
-        this.operatorGuid = operatorGuid;
-        this.isoLangCode = isoLangCode;
-        this.style = style;
-        this.langCodeId = LanguageStore.get(isoLangCode).getId();
-        this.formActivityDef = formActivityDef;
-        this.formResponse = formResponse;
+        this.params = params;
+        LanguageDto languageDto = LanguageStore.get(params.getIsoLangCode());
+        this.langCodeId = languageDto != null ? languageDto.getId() : 0;
```

Comment:
We should probably throw an error (or somehow return `Optional.empty()`) if it's an invalid language instead of defaulting to zero?

Reply:
It is checked at the beginning of the process (when parameters checked) and eventually Optional.empty() is returned:
```
if (languageDto == null) {
            context.setFailedMessage("Unknown language code: " + context.getParams().getIsoLangCode());
            context.setFailedStep(CHECK_PARAMS);
 } ...
```
And in old implementation it doesn't throw an exception when language code is unknown, therefore I did the same processing.
Or do you think that better to throw an exception in such case? (will need to change couple of tests then)

Reply:
Anyway I added check and exception throwing in AIBuilderContext constructor (double-checking)

Reply:
I think the old implementation assumes that this is non-null because the language should had been handled in `StudyLanguageResolutionFilter`. I think it'll be good to do similar thing here -- assume it's non-null instead of trying to fallback to something like zero. In my view, getting a NullPointerException will be better than having this silently go along with zero.

Reply:
Fixed